libr/core/anal_objc.c

```diff
@@ -182,23 +188,36 @@ static bool objc_find_refs(RCore *core) {
 
 	int total = 0;
 	ut64 off;
+	bool readSuccess = true;
 	for (off = 0; off < objc._data->vsize ; off += objc2ClassSize) {
```

Comment:
The root cause of this issue is that this for is assuming that `__objc_data` is uniformly filled with objc classes which is not true.

So it ends up reading pointers in the middle of other data types and the -1 is ultimately caused by `classRoVA` not being an actual address, but some random else.

Reply:
so maybe it's a good idea to just merge this to avoid the infinite loop, then fix the root cause for instance by iterating `__objc_classlist` and `__objc_catlist` instead and pull methods from there